src/west/manifest.py

```diff
@@ -203,21 +201,14 @@ def _load(self, data, sections):
 
         manifest = data.get('manifest')
 
-        try:
-            url = config.get('manifest', 'remote')
-            revision = config.get('manifest', 'revision')
-        except configparser.NoOptionError as e:
-            raise MalformedConfig('missing key: \'{}\' in west config file'
-                                  .format(e.args[0])) from e
-
-        name = posixpath.basename(urlparse(url).path)
+        path = config.get('manifest', 'path', fallback=None)
 
         self_tag = manifest.get('self')
-        path = self_tag.get('path') if self_tag else name
+        if path is None:
```

Comment:
We really need a `path` here or there is nothing more we can do. What about something like this?

```
if path is None:
    path = self_tag.get('path') if self_tag else None
if path is None:
    log.die('no manifest path and self.path is not set in the manifest file')
```

Reply:
`None` is ok.
In case this is first run in PostInit and a url was used for `west init` there is a chance that `self::path` is not present (which is perfectly legal situation).
In that case, `manifest.py` will create manifest project without a path, and then PostInit updates the path and name in the manifest project afterwards based on the url name.